### DISCUSSION TITLE
Only one unique index - validate my understanding pls

### DISCUSSION THREAD

**Discussion by iamjrock** [pgEdge/cli#52] [Q&A] [answered]
I have read about this limitation in the documentation [here](https://docs.pgedge.com/spock_ext/limitations#only-one-unique-indexconstraintpk) and I'd like to test my understanding with an example please.

If I'm using multi-active replication and my table structure is as follows....

```
customer_id (UUID) | customer_name (string) | email_address (string) 
```

...with these 2 indexes...

```
Primary Key: [customer_id]
Unique Index: [email_address]
```

...then this can break replication.

Correct?

If I've understood this correctly, then that would seem to make multi-active replication untenable for many use cases.  Are there any workarounds to this?  Something like EDB's [conflict triggers](https://www.enterprisedb.com/docs/pgd/latest/striggers/#conflict-triggers) etc?

**Comment by iamjrock** [comment]
Any thoughts on this are greatly appreciated please.

**Comment by luss** [comment, marked as answer]
yes, that could break replication.   It is recommended that you make your
second index non-unique.  The reason is that for conflict resolution there
must only be one possible unique key that can be used and can create
(potentially) a duplicate key on insert.

**Reply by iamjrock** [reply]
Ok thx for your insight on this, very much appreciated.  We can live without foreign keys etc, but not being able to have unique indexes, in addition to the primary key, is probably too large a sacrifice for our use case.

I imagine that it is an extremely complex, perhaps logically impossible, challenge to solve - but I'll pop the question anyway: Is there any roadmap to being able to safely run multi-master with multiple unique keys in a table?
